### ISSUE TITLE
False positive `declare-non-slot` on classvar

### ISSUE THREAD

**Issue by mthuurne** [pylint-dev/pylint#9950]
### Bug description

I'm checking code that simplifies to this testcase:

```python
from typing import ClassVar


class C:
    __slots__ = ("value",)

    def __init__(self, value: int):
        self.value = value

    x: ClassVar[int]
    y: ClassVar[int] = 2
    z = 3
```

For some reason, pylint considers `x` to be a declaration that conflicts with `__slots__`, but not `y` and `z`.

Curiously, this error flagged for `x` goes away if `__slots__` is empty.

### Command used

```shell
pylint non_slot_classvar.py
```

### Pylint output

```python
************* Module non_slot_classvar
non_slot_classvar.py:10:4: E0245: No such name 'x' in __slots__ (declare-non-slot)
```

### Expected behavior

No error should be flagged.

At runtime, there is no error, as `__slots__` only applies to instances of `C`, not to the class.

### Pylint version

```shell
pylint 3.3.0
astroid 3.3.3
Python 3.11.9 (main, Apr 08 2024, 06:18:15) [GCC]
```

### OS / Environment

Linux (openSUSE Tumbleweed)

**Comment by mthuurne** [comment]
This problem is new since pylint 3.3.0, which makes sense as `E0245` was added recently, in #9564.

**Comment by jacobtylerwalls** [comment]
cc/ @adamtuft, could you lend us your view of this bug report? Thanks!

**Comment by adamtuft** [comment]
Thanks for tagging me, I'll take a look 🔍
